### PR TITLE
feat: add base settings page

### PR DESCRIPTION
**Pull request by rkorzhoff**
## Summary by CodeRabbit

- **New Features**
	- Added a settings page for managing application configurations such as OpenAI API key, Python interpreter path, and optional agent settings.
	- Introduced a new approval style for buttons to enhance user interaction.

- **Enhancements**
	- Implemented functionality to retrieve and update settings via API calls, improving the application's flexibility and user experience.

- **UI Updates**
	- Integrated a `SettingsForm` component for a streamlined settings management experience.
	- Updated the settings page layout to include form fields and action buttons for a user-friendly interface.

### CODE REVIEW

File: entities/settings/model/Settings.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export interface Settings {
+  openai_api_key: string
+  python_path: string
+  agents?: unknown
```
[the remark this answers is not in the document]
Comment (rkorzhoff) [reply entities/settings/model/Settings.ts:4]: We will define it properly later

File: features/settings/model/UpdateSettings.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export interface UpdateSettings {
+  openai_api_key?: string
+  python_path?: string
+  agents?: unknown
```
[the remark this answers is not in the document]
Comment (rkorzhoff) [reply features/settings/model/UpdateSettings.ts:4]: Same here

File: features/settings/api/getSettings.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2024 StarfleetAI
+// SPDX-License-Identifier: Apache-2.0
+
+import type { Settings } from '~/entities/settings'
+
+export const getSettings = () => {
+  return invoke<Settings>('get_settings')
+}
```
[the remark this answers is not in the document]
Comment (rkorzhoff) [reply features/settings/api/getSettings.ts:8]: Error handlers will be implemented later

File: features/settings/api/updateSettings.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2024 StarfleetAI
+// SPDX-License-Identifier: Apache-2.0
+
+import type { UpdateSettings } from '../model'
+
+export const updateSettings = (newSettings: UpdateSettings) => {
+  return invoke('update_settings', { newSettings })
+}
```
[the remark this answers is not in the document]
Comment (rkorzhoff) [reply features/settings/api/updateSettings.ts:8]: Error handlers will be implemented later

File: widgets/settingsForm/ui/SettingsForm.vue [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  }
+  const handleSave = async () => {
+    await updateSettings({
+      openai_api_key: changedSettings.value.openai_api_key,
+      python_path: changedSettings.value.python_path,
+      agents: changedSettings.value.agents,
+    })
+
+    settings.value = await getSettings()
+  }
```
[the remark this answers is not in the document]
Comment (rkorzhoff) [reply widgets/settingsForm/ui/SettingsForm.vue:30]: Error handlers will be implemented later

**Review by botjoker — APPROVED** [review]

**Pull request merged**